src/validation/rules/OverlappingFieldsCanBeMergedRule.ts

```diff
@@ -830,28 +832,27 @@ function getStreamDirective(
   return directives.find((directive) => directive.name.value === 'stream');
 }
 
-function sameStreams(
+function hasNoOverlappingStreams(
   directives1: ReadonlyArray<DirectiveNode>,
   varMap1: Map<string, ValueNode> | undefined,
   directives2: ReadonlyArray<DirectiveNode>,
   varMap2: Map<string, ValueNode> | undefined,
-): boolean {
+): string | undefined {
   const stream1 = getStreamDirective(directives1);
   const stream2 = getStreamDirective(directives2);
   if (!stream1 && !stream2) {
     // both fields do not have streams
-    return true;
+    return;
   } else if (stream1 && stream2) {
     // check if both fields have equivalent streams
-    return sameArguments(
-      stream1.arguments,
-      varMap1,
-      stream2.arguments,
-      varMap2,
-    );
+    if (sameArguments(stream1.arguments, varMap1, stream2.arguments, varMap2)) {
+      // This was allowed in previous alpha versions of `graphql-js`.
+      return 'they have differing stream directives. See https://github.com/graphql/defer-stream-wg/discussions/100';
```

Comment:
nit: this seemingly should be 'overlapping stream directives', we don't know if they differ in terms of selection sets.

Reply:
@yaacovCR make sense, updated.